File: Easy/Others/Reversebits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int[] bitArr = new int[32];
        Arrays.fill(bitArr,0);
        int i = bitArr.length-1;
        while(n!=0)
        {
        	bitArr[i] = n%2;
        	n = n/2;
        	i--;
        }
        int result = 0;
        for(int j=0;j<bitArr.length;j++)
        {
        	result = result + bitArr[j]*(int)Math.pow(2,j);
        }
        return result;
    }
}

File: CSharp/Easy/Other/ReverseBits.cs
/*
Example:

Input: 43261596
Output: 964176192
Explanation: 43261596 represented in binary as 00000010100101000001111010011100, 
             return 964176192 represented in binary as 00111001011110000010100101000000.
*/
public class Solution {
    public uint reverseBits(uint n) {
        uint result = 0;
        uint index=31;
        while(n!=0)
        {
        	result = result + ((n%2)*( (uint)Math.Pow(2,index) ));
        	n = n/2;
        	index--;
        }
        
        return result;
    }
}